src/System.Memory/tests/ReadOnlyBuffer/ReadOnlySequenceFactory.byte.cs

```diff
@@ -103,32 +102,25 @@ public override ReadOnlySequence<T> CreateWithContent(T[] data)
             }
         }
 
-        public static ReadOnlySequence<T> CreateSegments(params T[][] inputs)
+        public static ReadOnlySequence<T> CreateSegments(params T[][] inputs) => CreateSegments((IEnumerable<T[]>)inputs);
+
+        public static ReadOnlySequence<T> CreateSegments(IEnumerable<T[]> inputs)
         {
-            if (inputs == null || inputs.Length == 0)
+            if (inputs == null || inputs.Count() == 0)
             {
                 throw new InvalidOperationException();
             }
 
-            int i = 0;
-
             BufferSegment<T> last = null;
             BufferSegment<T> first = null;
-
-            do
+            foreach(T[] input in inputs)
             {
-                T[] s = inputs[i];
-                int length = s.Length;
-                int dataOffset = length;
-                var chars = new T[length * 2];
-
-                for (int j = 0; j < length; j++)
-                {
-                    chars[dataOffset + j] = s[j];
-                }
-
-                Memory<T> memory = new Memory<T>(chars).Slice(length, length);
-
+                int length = input.Length;
+                int dataOffset = length / 2;
```

Comment:
**Question:** This used to be `dataOffset = length`, why change it to `length / 2`?

Reply:
Would it be better to move tested to middle of array? Or stay them at end?

Reply:
It is fine as is. Thanks.